preprocessing/nextclade/src/loculus_preprocessing/processing_functions.py

```diff
@@ -198,6 +201,163 @@ def check_date(
+                    f"new upper limit: {max_upper_limit}"
+                )
+                datum.date_range_upper = max_upper_limit
+
+            if message:
+                warnings.append(
+                    ProcessingAnnotation(
+                        source=[
+                            AnnotationSource(name=output_field, type=AnnotationSourceType.METADATA)
+                        ],
+                        message=f"Metadata field {output_field}:'{date_str}' - " + message,
+                    )
+                )
+
+            if datum.date_range_lower > datetime.now(tz=pytz.utc):
+                logger.debug(
+                    f"Lower range of date: {datum.date_range_lower} > {datetime.now(tz=pytz.utc)}"
+                )
+                errors.append(
+                    ProcessingAnnotation(
+                        source=[
+                            AnnotationSource(name=output_field, type=AnnotationSourceType.METADATA)
+                        ],
+                        message=f"Metadata field {output_field}:'{date_str}' is in the future.",
+                    )
+                )
+
+            if release_date and (datum.date_range_lower > release_date):
+                logger.debug(f"Lower range of date: {parsed_date} > release_date: {release_date}")
+                errors.append(
+                    ProcessingAnnotation(
+                        source=[
+                            AnnotationSource(name=output_field, type=AnnotationSourceType.METADATA)
+                        ],
+                        message=(
+                            f"Metadata field {output_field}:'{date_str}'" "is after release date."
+                        ),
+                    )
+                )
+
+            match args["fieldType"]:
+                case "dateRangeString":
+                    return_value = datum.date_range_string
+                case "dateRangeLower":
+                    return_value = datum.date_range_lower.strftime("%Y-%m-%d")
+                    warnings = errors = []
+                case "dateRangeUpper":
+                    return_value = datum.date_range_upper.strftime("%Y-%m-%d")
+                    warnings = errors = []
+                case _:
+                    msg = f"Config error: Unknown fieldType: {args['fieldType']}"
+                    raise ValueError(msg)
+
+            return ProcessingResult(datum=return_value, warnings=warnings, errors=errors)
+
+        # If all parsing attempts fail, it's an unrecognized format
+        return ProcessingResult(
+            datum=None,
+            warnings=[],
+            errors=[
+                ProcessingAnnotation(
+                    source=[
+                        AnnotationSource(name=output_field, type=AnnotationSourceType.METADATA)
+                    ],
+                    message=f"Metadata field {output_field}: Date format is not recognized.",
```

Comment:
Does it make sense to include the erroneous string in the message as well?

Reply:
Any input on this? To see the incorrectly formatted input in the error message helps the user narrow down where they made a mistake.